src/encoded/static/components/objectutils.js

```diff
@@ -40,6 +40,27 @@ DisplayAsJson.contextTypes = {
     location_href: PropTypes.string,
 };
 
+export function findRelatedLocation(context, assembly) {
+    let relatedDatasetsLocation = null;
+    if (context.related_datasets) {
+        for (let idx = 0; idx < context.related_datasets.length; idx += 1) {
+            if (context.related_datasets[idx].elements_references) {
+                for (let jdx = 0; jdx < context.related_datasets[idx].elements_references.length; jdx += 1) {
+                    if (context.related_datasets[idx].elements_references[jdx].examined_loci) {
+                        for (let kdx = 0; kdx < context.related_datasets[idx].elements_references[jdx].examined_loci.length; kdx += 1) {
+                            if (context.related_datasets[idx].elements_references[jdx].examined_loci[kdx].locations && context.related_datasets[idx].elements_references[jdx].examined_loci[kdx].locations.find((elem) => elem.assembly === assembly)) {
+                                relatedDatasetsLocation = context.related_datasets[idx].elements_references[jdx].examined_loci[kdx].locations.find((elem) => elem.assembly === assembly);
+                                return relatedDatasetsLocation;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return null;
+}
```

Comment:
I feel like this isn’t easy to maintain, and it potentially executes the same `.find()` loop twice. I didn’t test whether this works, but I think this is a lot easier and has a lot less duplication:
```
export function findRelatedLocation(context, assembly) {
    if (context.related_datasets.length > 0) {
        for (const relatedDataset of context.related_datasets) {
            if (related_dataset.elements_references?.length > 0) {
                for (const elementsReference of relatedDataset.elements_references) {
                    if (elementsReference.examined_loci?.length > 0) {
                        for (const examinedLoci of elementsReference.examined_loci) {
                            if (examinedLoci.locations?.length > 0) {
                                const relatedDatasetsLocation = examinedLoci.find(
                                    (elem) => elem.assembly === assembly
                                );
                                if (relatedDatasetsLocation) {
                                    return relatedDatasetsLocation;
                                }
                            }
                        }
                    }
                }
            }
        }
    }
    return null;
}
```
What’s your rule for deciding whether to use a function declaration vs a function expression? I think Phil asked you that once but I didn’t pay attention to the conversation after that.

Reply:
This is a great way to write the function, thank you for suggesting it! I agree that executing the find() just once is much better and I like the briefer syntax. I wrote this function the way that most of other functions were written in objectutils but if the other way would be better I can change it!

Reply:
I got an error when I tried this approach:

> iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations  no-restricted-syntax

So I tried using .some() because maybe that is better than the regular for loop.
Let me know what you think!

Reply:
Interesting. It sounds like that error is from an AirBnB rule that people [debate](https://github.com/airbnb/javascript/issues/1271) the value of. But if what you have works, might as well stick with that. Maybe AirBnB has the right idea for reasons I’m not aware of.